sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -373,7 +372,9 @@ private void addTimedEvent(Event timedEvent) {
         logger.log(Level.FINE, "Calling addEvent() on an ended Span.");
         return;
       }
-      events.add(timedEvent);
+      if (events.size() < traceConfig.getMaxNumberOfEvents()) {
```

Comment:
Spec just says discard after reaching the limit, which is closer to what this code does than the previous FIFO queue (though personally I think either is fine)

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#limits-on-span-collections